rasa/nlu/server.py

```diff
@@ -111,258 +139,221 @@ def dump_to_data_file(data):
+            data = model_config.get("data")
 
         else:
-
             raise Exception(
                 "Content-Type must be 'application/x-yml' or 'application/json'"
             )
 
         return model_config, data
 
-    def get_request_content_type(self, request):
-        content_type = request.requestHeaders.getRawHeaders("Content-Type", [])
-
-        if len(content_type) is not 1:
-            raise Exception("The request must have exactly one content type")
-        else:
-            return content_type[0]
-
-    @app.route("/train", methods=["POST", "OPTIONS"], branch=True)
-    @requires_auth
-    @check_cors
-    @inlineCallbacks
-    def train(self, request):
+    @app.post("/train")
+    @requires_auth(app, token)
+    async def train(request):
         # if not set will use the default project name, e.g. "default"
-        project = parameter_or_default(request, "project", default=None)
+        project = request.args.get("project", None)
         # if set will not generate a model name but use the passed one
-        model_name = parameter_or_default(request, "model", default=None)
+        model_name = request.args.get("model", None)
 
         try:
-            model_config, data = self.extract_data_and_config(request)
-
+            model_config, data = extract_data_and_config(request)
         except Exception as e:
-            request.setResponseCode(400)
-            returnValue(json_to_string({"error": "{}".format(e)}))
+            raise ErrorResponse(
+                500,
+                "Server Error",
+                "An unexpected error occurred.",
+                details={"error": str(e)},
+            )
 
         data_file = dump_to_data_file(data)
 
-        request.setHeader("Content-Type", "application/zip")
-
         try:
-            request.setResponseCode(200)
-            request.setHeader("Content-Disposition", "attachment")
-            path_to_model = yield self.data_router.start_train_process(
+            path_to_model = await data_router.start_train_process(
                 data_file, project, RasaNLUModelConfig(model_config), model_name
             )
             zipped_path = utils.zip_folder(path_to_model)
-
-            zip_content = io.open(zipped_path, "r+b").read()
-            return returnValue(zip_content)
+            return await response.file(zipped_path)
 
         except MaxWorkerProcessError as e:
-            request.setResponseCode(403)
-            returnValue(json_to_string({"error": "{}".format(e)}))
+            raise ErrorResponse(
+                403,
+                "Forbidden",
+                "No process available for training.",
+                details={"error": str(e)},
+            )
         except InvalidProjectError as e:
-            request.setResponseCode(404)
-            returnValue(json_to_string({"error": "{}".format(e)}))
+            raise ErrorResponse(
+                404, "Not Found", "No project found.", details={"error": str(e)}
```

Comment:
let's include the invalid project's name 
```suggestion
                404, "Project Not Found", "Project '{}' not found.".format(project), details={"error": str(e)}
```